shared/chat/conversation/input-area/normal/container.tsx

```diff
@@ -205,7 +207,20 @@ export default Container.namedConnect(
       dispatch(Chat2Gen.createSetExplodingModeLock({conversationIDKey, unset})),
   }),
   (stateProps, dispatchProps, ownProps: OwnProps) => {
+    const botRestrictMap = stateProps._botSettings
```

Comment:
@chrisnojima Thoughts about having this here? I realize it could be a perf impact (although low), but it is convenient to pick up changes from a bunch of sources that could affect the end result.